lib/puppet/application/lookup.rb

```diff
@@ -339,6 +339,7 @@ def generate_scope
 
     Puppet[:code] = 'undef' unless options[:compile]
     compiler = Puppet::Parser::Compiler.new(node)
+    compiler.loaders # ensures that Pcore et.al. is initialized
```

Comment:
No, it doesn't. The only code-path where this problem can occur is if ruby code is trying to access the loader without any pp code has been loaded. Catalog preview will not do that since it doesn't call lookup by itself.